RNA-seq/00c-tximeta_other_species.Rmd

```diff
@@ -0,0 +1,267 @@
+
+### Salmon index 
+
+We need to provide `tximeta` with the salmon index for the species and genome release we are using. 
+We had prepared our salmon index for zebrafish along with other species which you can find more information about [on GitHub](https://github.com/AlexsLemonade/training-txome-prep).
+If you prepare your own salmon index, just be sure to note what database and release you used to build it. 
+
+```{r directories}
+# Set up file path to the salmon index for the species we are using
+index_dir <- file.path("index", "Danio_rerio", "short_index")
+```
+
+### FASTA sequence files
+
+`tximeta` will need the FASTA files for matching the transcript sequences to our data. 
+These FASTA files should match whatever was used to build your salmon index and can be obtained from Ensembl's ftp server. 
+
+You can see the full list of species and FASTA file options (should you be using a different species): http://ftp.ensembl.org/pub/release-95/fasta/
+
+If you using a species other than zebrafish, click through the file paths on the [ftp server](http://ftp.ensembl.org/pub/release-95/fasta/) to find the corresponding files for your species of interest. 
+
+The `base_url` is where all Ensembl's species FASTA files for release 95 are located. 
+This will not need to change should you use another species also from Ensembl's release 95, but if you need a different release, the `release-95` part will need to be changed accordingly. 
+
+```{r}
+# This base URL will not need to change for other species of the same release
+fasta_base_url <- "ftp://ftp.ensembl.org/pub/release-95/fasta/"
+```
+
+The second part of this URL is species-specific which will require you to update to your species' equivalent file path. 
+Note that the FASTA files used here should be the same as what you used to build your salmon index. 
+If you used multiple FASTA files to build your index, you can make this into a vector and include multiple FASTA files here as well. 
+
+```{r}
+fasta_urls <- paste0(
+  fasta_base_url, 
+  # Replace the file path below to the species-specific fasta cDNA file
+  "danio_rerio/cdna/Danio_rerio.GRCz11.cdna.all.fa.gz" 
+  )
+```
+
+### GTF file
+
+Now we can follow a similar series of steps to set up the GTF file's URL for our species using this address: 
+http://ftp.ensembl.org/pub/release-95/gtf/
+
+Again, if you are using a different release, the `release-95` part below will need to be altered. 
+
+```{r}
+# This base URL will not need to change for other species of the same release
+gtf_base_url <- "ftp://ftp.ensembl.org/pub/release-95/gtf/"
+```
+
+Similarly, the second part of this URL will need to be replaced with your species' equivalent file path (if you are not using zebrafish). 
+
+```{r}
+gtf_url <- paste0(
+  gtf_base_url, 
+  # Replace the file path below to the species-specific gtf file
+  "danio_rerio/Danio_rerio.GRCz11.103.gtf.gz"
```

Comment:
Why did this still work? (Or did it not and tximeta still ran anyway)?

Reply:
Not sure, because I didn't test it wrong, but my guess (from some of the status messages) is that it doesn't actually download the gtf if it can find the ensembl annotation through `AnnotationDbi`. I tried to see what would happen if  I left off fasta and/or gtf files, and that failed at the `makeLinkedTxome` step, but for the full case, when you actually get to `tixmeta` it says something about downloading from AnnotationDbi, so I think it shortcuts when it can.

Reply:
SOooo even after the url is changed to release-95, `tximeta()` pulls from a LinkedTxome that still has mixed releases: 
<img width="866" alt="Screen Shot 2021-02-25 at 8 14 42 AM" src="https://user-images.githubusercontent.com/23458084/109158670-ad5c0280-7741-11eb-9163-b12907e9be70.png">

Next, I tried clearing all the LinkedTxomes and re-creating one that has the files with all the releases matching. But in a weirder twist, it now does not recognize that the created matching-release linkedTxome for the current zebrafish dataset. So I'm not sure how it determines what linkedTxome works for what dataset. 

Going to dig a tad more because this is weird.

Reply:
Figured it out with jashapiro's help. The salmon_index in my own directory was the wrong species leftover from when I was editing `setup/link-data.sh` and had an error in it. Though I had re-run `link-data.sh` that wasn't enough to fix it because it needed to be deleted. So basically I was sending the `linkedTxome` to a `Homo_sapiens` index that I thought was `Danio_rerio`. Once I fixed that and re-made the `linkedTxome` now `tximeta` runs properly. 

Bigger picture lessons learned from this about tximeta: 1) the linkedTxome won't complain if the species are different in the files you provide it. 2) tximeta might run with a linkedTxome that has different releases if you have one that matches.